server/data/previewClient.ts

```diff
@@ -0,0 +1,26 @@
+import type { ApiConfig } from '@ministryofjustice/hmpps-digital-prison-reporting-frontend/dpr/components/report-list/data/types'
+import RestClient from './restClient'
+
+export default class PreviewClient {
+  private config: ApiConfig
+
+  constructor(config: ApiConfig) {
+    this.config = config
+  }
+
+  deleteDefinition(definitionId: string, token: string): Promise<void> {
+    return new RestClient('Preview API Client', this.config, token).delete({
+      path: `/definitions/${definitionId}`,
+    })
+  }
+
+  uploadDefinition(definition: string, token: string): Promise<void> {
+    return new RestClient('Preview API Client', this.config, token).put({
+      path: `/definitions/definitionId`,
```

Comment:
Question: Should the definitionId here be defined as ${definitionId} similar to the DELETE endpoint? 
As it is currently it seems like a string.

Reply:
Yes, I've been a bit cheeky here. To get the definition ID out of the definition I'd have to parse it, whereas currently it's just a pass-through. The definition ID isn't actually used by the API though, at the moment.